diff --git a/src/slirp.h b/src/slirp.h
--- a/src/slirp.h
+++ b/src/slirp.h
@@ -218,6 +218,7 @@ struct Slirp {
     struct sockaddr_in *outbound_addr;
     struct sockaddr_in6 *outbound_addr6;
     bool disable_dns; /* slirp will not redirect/serve any DNS packet */
+    bool disable_dhcp; /* slirp will not reply to any DHCP requests */
     char *outbound_proxy;
     void *proxy_state;
 };
diff --git a/src/proxy.c b/src/proxy.c
--- a/src/proxy.c
+++ b/src/proxy.c
@@ -366,8 +366,6 @@ static void proxy_complete(Slirp *slirp)
             slirp_register_poll_socket(so);
             so->so_state &= ~SS_ISFCONNECTING;
         } else {
-            curl_easy_cleanup(easy);
-            so->proxy_handle = NULL;
             so->s = SLIRP_INVALID_SOCKET;
             so->so_state &= SS_PERSISTENT_MASK;
             so->so_state |= SS_NOFDREF;
